### ISSUE TITLE
The mouse mode selector can become inaccessible after window resize

### ISSUE THREAD

**Issue by petrcermak** [google/trace-viewer#941]
The figure below demonstrates the problem:

![mouse_selector_problem](https://cloud.githubusercontent.com/assets/2546601/7429210/b8a42ebc-eff5-11e4-8ebf-13599a948e57.png)

**Comment by petrcermak** [comment]
I dug through the code and found the following:

1. **All changes to the timeline view** (including *window resize* and opening/closing the *side pane*) **except for analysis view resize are routed through <tt>TimelineView.onViewportChanged_()</tt>**. Since this class has a reference to the <tt>TimelineTrackView</tt>, which in turn has a reference to the <tt>MouseModeSelector</tt>, propagating the change to the selector should be relatively straightforward (timeline view → timeline track view → mouse mode selector).
2. **Resizing the analysis view** (caused by <tt>DragHandle.applyDelta_</tt>) **does not seem to trigger any hooks on any of the timeline classes**. Propagating the change to the mouse mode selector would, therefore, probably be a little more complex. The good news is that resizing the analysis view can *never* cause the selector to become inaccessible (all it can do is to make the selector appear above the analysis view instead of the timeline) so the issue is "only" a visual glitch in this case (and hence is lower priority).

To sum up, I think that we should propagate the change from <tt>TimelineView.onViewportChanged_()</tt> to <tt>MouseModeSelector</tt> (first point above), which would address the potential inaccessibility of the selector.

@natduca, @dj2: WDYT?

_If only there were a <tt>resize</tt> event on arbitrary HTML elements..._

**Comment by natduca** [comment]
Another option is to ditch the hover entirely and move it to the top, a la #868

**Comment by natduca** [comment]
As far as the viewportchanged thing goes, i'd really rather us not hook to viewport changed. Thats about the zoom/viewport/etc system, and only accidentally is broadcast when the track view resizes.

Mouse mode selector really wants to observe when its parent changes size. What we really want is for elements to send a resize event and then we reposition on that event.

https://github.com/marcj/css-element-queries has a ResizeSensor.js in it. I wonder if we had that in third_party, and then had the mms use that....

**Comment by petrcermak** [comment]
I guess the best option might be to indeed *get rid of the selector completely*. But I agree on the resize events.
